pennylane/devices/qubit/simulate.py

```diff
@@ -124,24 +134,34 @@ def get_final_state(circuit, debugger=None, interface=None):
 
     # initial state is batched only if the state preparation (if it exists) is batched
     is_state_batched = bool(prep and prep.batch_size is not None)
+    measurement_values = {}
     for op in circuit.operations[bool(prep) :]:
-        state = apply_operation(op, state, is_state_batched=is_state_batched, debugger=debugger)
-
+        if isinstance(op, Conditional):
+            meas_id = op.meas_val.measurements[0].hash
+            if not measurement_values[meas_id]:
```

Comment:
```suggestion
            if not measurement_values[op.meas_val.measurements[0]]:
```

Since `op.meas_val.measurments[0]` is a `MidMeasureMP`, we should be able to hash on the value itself.  This has the benefit of making the `measurement_values` itself easier to read and debug, since the keys are the actual object, not a long number.

https://github.com/PennyLaneAI/pennylane/blob/127beeb6ccf99f0ac7859c6e3358eb034b1a425e/pennylane/measurements/measurements.py#L303

Reply:
To avoid keeping track of too many such values in the composite case, I introduced a `concretize` method in MVs (in #5120) which takes in a `dict` of the MCMs and recursively figures out the concrete value of a measurement or composite measurement. This ends up looking like [that](https://github.com/PennyLaneAI/pennylane/pull/5120/files#diff-a5e8e7161dde1247b6305745302f216f95e68ea8e42ceb7dea59bae63cf2b263R140). If you think it's ok, we could merge as is and reevaluate in the follow-up PR.